### DISCUSSION TITLE
How GeckoLinux builds Spiral

### DISCUSSION THREAD

**Discussion by methanoid** [SpiralLinux/SpiralLinux-project#408] [Q&A]
Hi

I've asked questions before (and then been quiet for a while) so I know how to use LiveBuild etc and your configs are available 

My question is more how does @geckolinux  prepare the config files and/or batch build the ISOs?

I'll explain whY I am being so nosy - I LOVE Spiral - I love the sane defaults and lots of things about it but like 99% of Linux users I want to make it my own. Yes, I could script it all post install but I'd like to build my own ISOs en-masse for Plasma/Gnome/XFCE/Cinnamon/Budgie (not interested in other Spirals). I'm sure Gecko has the workflow optimized to build the configs and the ISOs so why not ask someone who does it regularly? :-)

Finally (unrelated). Is it possible to add Pantheon and Deepin DE versions of Spiral and is Cosmic desktop being considered (as an alpha or beta Spiral)?

No matter how much I try other distros, I keep finding Spiral's "sane Debian" is the one for me

**Comment by geckolinux** [comment]
Hi there @methanoid , glad to hear that SpiralLinux is working well for your needs.

Here is a fairly detailed description about how I build the SpiralLinux ISOs:
-  https://github.com/orgs/SpiralLinux/discussions/321#discussioncomment-7235501
- https://github.com/orgs/SpiralLinux/discussions/228#discussioncomment-4809266

The config files are generated by simply booting up the SpiralLinux live ISO in a virtual environment and changing the settings the way that one would normally do for a personal installation, and then I create an SFTP connection into the virtual machine and copy the relevant config files into the SpiralLinux overlay files. Sometimes there are certain config options (relating to touchpads and mouse scrolling) that only appear on real hardware, so I boot the SpiralLinux live ISO on a test laptop and make those settings and then copy over the modified config files to my SpiralLinux directory.

> Finally (unrelated). Is it possible to add Pantheon and Deepin DE versions of Spiral and is Cosmic desktop being considered (as an alpha or beta Spiral)?

Unfortunately no, or at least not yet, as those desktops are currently not packaged for Debian.
